indy-utils/src/main/kotlin/com/luxoft/blockchainlab/hyperledger/indy/IndyFacade.kt

```diff
@@ -0,0 +1,193 @@
+        proverDid: String,
+        offer: CredentialOffer,
+        masterSecretId: String = DEFAULT_MASTER_SECRET_ID
+    ): CredentialRequestInfo
+
+    /**
+     * Issues [Credential] by [CredentialRequest] and [CredentialOffer] using [WalletService].
+     * If revocation is enabled it will hold one of [maxCredentialNumber].
+     *
+     * @param credentialRequest [CredentialRequestInfo] - [CredentialRequest] and all reliable info
+     * @param offer [CredentialOffer] - credential offer
+     * @param revocationRegistryId [RevocationRegistryDefinitionId] or [null] - revocation registry definition id
+     * @param proposalProvider lambda returning [Map] of [String] to [CredentialValue] - credential proposal
+     *
+     * @return [CredentialInfo] - credential and all reliable data
+     */
+    fun issueCredentialAndUpdateLedger(
+        credentialRequest: CredentialRequestInfo,
+        offer: CredentialOffer,
+        revocationRegistryId: RevocationRegistryDefinitionId?,
+        proposalProvider: () -> CredentialProposal
+    ): CredentialInfo
+
+    /**
+     * Stores [Credential] in prover's wallet gathering data using [LedgerService]
+     *
+     * @param credentialInfo [CredentialInfo] - credential and all reliable data
+     * @param credentialRequest [CredentialRequestInfo] - credential request and all reliable data
+     * @param offer [CredentialOffer]
+     */
+    fun receiveCredential(
+        credentialInfo: CredentialInfo,
+        credentialRequest: CredentialRequestInfo,
+        offer: CredentialOffer
+    )
+
+    /**
+     * Revokes previously issued [Credential] using [WalletService] and [LedgerService]
+     *
+     * @param revocationRegistryId [RevocationRegistryDefinitionId] - revocation registry definition id
+     * @param credentialRevocationId [String] - revocation registry credential index
+     */
+    fun revokeCredentialAndUpdateLedger(
+        revocationRegistryId: RevocationRegistryDefinitionId,
+        credentialRevocationId: String
+    ): RevocationRegistryEntry
+
+    /**
+     * Creates [ProofRequest]. This function has nothing to do with Indy API, it is used just to produce well-shaped data.
+     *
+     * @param version [String] - ???
+     * @param name [String] - name of this proof request
+     * @param attributes [List] of [CredentialFieldReference] - attributes which prover needs to reveal
+     * @param predicates [List] of [CredentialPredicate] - predicates which prover should answer
+     * @param nonRevoked [Interval] or [null] - time interval of [attributes] and [predicates] non-revocation
+     * @param nonce [String] - random value to distinct identical proofs
+     *
+     * @return [ProofRequest]
+     */
+    fun createProofRequest(
+        version: String,
+        name: String,
+        attributes: List<CredentialFieldReference>,
+        predicates: List<CredentialPredicate>,
+        nonRevoked: Interval?,
+        nonce: String = "123123"
```

Comment:
May be it should be random?

Reply:
It is random in the upcoming PR (test-coverage)